src/bloqade/analog/task/exclusive.py

```diff
@@ -0,0 +1,347 @@
+        response_data = response.json()
+        # Get "matched" from the response
+        matches = response_data.get("matches", None)
+        # The return is a list of dictionaries
+        # Verify if the list contains only one element
+        if matches is None:
+            print("No task found with the given ID.")
+            return "Failed"
+        elif len(matches) > 1:
+            print("Multiple tasks found with the given ID.")
+            return "Failed"
+
+        # Extract the status from the first dictionary
+        status = matches[0].get("status")
+        return status
+
+
+    def fetch_results(self, task_id: str):
+        response = request(
+            "GET",
+            self.verrcel_api_url,
+            params={
+                "searchPattern": task_id,
+                "magicToken": self.zapier_webhook_key,
+                "useRegex": False,
+            },
+        )
+        if response.status_code != 200:
+            print(
+                f"HTTP request failed with status code: {response.status_code}")
+            print("HTTP responce: ", response.text)
+            return None
+
+        response_data = response.json()
+        # Get "matched" from the response
+        matches = response_data.get("matches", None)
+        # The return is a list of dictionaries
+        # Verify if the list contains only one element
+        if matches is None:
+            print("No task found with the given ID.")
+            return None
+        elif len(matches) > 1:
+            print("Multiple tasks found with the given ID.")
+            return None
+        record = matches[0]
+        if record.get("status") == "Completed":
+            googledoc = record.get("resultsFileUrl")
+
+            # convert the preview URL to download URL
+            googledoc = convert_preview_to_download(
+                googledoc)
+            res = get(googledoc)
+            res.raise_for_status()
+            data = res.json()
+
+            task_results = QuEraTaskResults(**data)
+        return task_results
+
+
+class TestHTTPHandler(HTTPHandlerABC):
+    pass
+
+@Serializer.register
+class ExclusiveRemoteTask(CustomRemoteTaskABC):
+    def __init__(
+        self,
+        task_ir: QuEraTaskSpecification,
+        metadata: Dict[str, ParamType],
+        parallel_decoder: ParallelDecoder | None,
+        http_handler: HTTPHandlerABC | None = HTTPHandler(),
+        task_id: str = None,
+        task_result_ir: QuEraTaskResults = None,
+    ):
+        self._http_handler = http_handler 
+        self._task_ir = task_ir
+        self._metadata = metadata
+        self._parallel_decoder = parallel_decoder
+        float_sites = list(
+            map(lambda x: (float(x[0]), float(x[1])), task_ir.lattice.sites)
+        )
+        self._geometry = Geometry(
+            float_sites, task_ir.lattice.filling, parallel_decoder
+        )
+        self._task_id = task_id
+        self._task_result_ir = task_result_ir
```

Comment:
As per our discussion 1 on 1, use a dataclass to clean up this implementation, Also you should use default_factories for the default values instead of instances of objects that are mutable.